diff --git a/dist/runtime/components/Input.vue b/dist/runtime/components/Input.vue
index c61cdbb49b8e413def50fc18b393983e989e2aff..8f7d0d7c35aaf604ca53cba0eb7cca21989f5abd 100644
--- a/dist/runtime/components/Input.vue
+++ b/dist/runtime/components/Input.vue
@@ -80,7 +80,20 @@ function updateInput(value) {
   modelValue.value = value;
   emitFormInput();
 }
+
+let isComposing = false;
+
+function onCompositionStart() {
+  isComposing = true;
+}
+
+function onCompositionEnd(event) {
+  isComposing = false;
+  onInput(event);
+}
+
 function onInput(event) {
+  if (isComposing) return;
   if (!props.modelModifiers?.lazy) {
     updateInput(event.target.value);
   }
@@ -131,6 +144,8 @@ defineExpose({
       :autocomplete="autocomplete"
       v-bind="{ ...$attrs, ...ariaAttrs }"
       @input="onInput"
+      @compositionstart="onCompositionStart"
+      @compositionend="onCompositionEnd"
       @blur="onBlur"
       @change="onChange"
       @focus="emitFormFocus"
